front/components/assistant/AssistantDetailsDropdownMenu.tsx

```diff
@@ -25,40 +26,39 @@ import { useAgentConfiguration } from "@app/lib/swr/assistants";
 import { useUser } from "@app/lib/swr/user";
 
 interface AssistantDetailsDropdownMenuProps {
-  agentConfigurationId: string;
+  agentConfiguration: LightAgentConfigurationType;
   owner: WorkspaceType;
   variant?: "button" | "plain";
-  onAgentDeletion?: () => void;
+  canDelete?: boolean;
+  showAssistantDetails?: () => void;
   showAddRemoveToList?: boolean;
 }
 
 export function AssistantDetailsDropdownMenu({
-  // The `agentConfiguration` cannot be used directly as it isn't dynamically
-  // updated upon user list mutations. This limitation stems from its
-  // propagation method from <ConversationMessage>.
-  agentConfigurationId,
+  agentConfiguration,
   owner,
   variant = "plain",
-  onAgentDeletion,
+  canDelete,
+  showAssistantDetails,
   showAddRemoveToList = false,
 }: AssistantDetailsDropdownMenuProps) {
   const [isUpdatingList, setIsUpdatingList] = useState(false);
   const sendNotification = useContext(SendNotificationsContext);
   const { user } = useUser();
-  const { agentConfiguration, mutateAgentConfiguration } =
-    useAgentConfiguration({
-      workspaceId: owner.sId,
-      agentConfigurationId,
-    });
+  const { mutateAgentConfiguration } = useAgentConfiguration({
+    workspaceId: owner.sId,
+    agentConfigurationId: agentConfiguration.sId,
+    disabled: true,
```

Comment:
@spolu the diff with the reverted PR is here.
I'm disabling the hook (but keep it for the mutate) and using directly the configuration.